subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AbstractClassGenerator.java

```diff
@@ -733,9 +734,10 @@ private Method findClosureOverload(Method method, Collection<Method> candidates)
                     continue;
                 }
                 boolean matches = true;
-                for (int i = 0; matches && i < candidate.getParameterTypes().length - 1; i++) {
+                for (int i = 0; i < candidate.getParameterTypes().length - 1; i++) {
```

Comment:
May be unrelated, but it seems to me this method could probably be simplified to something like:
```java
@Nullable
private static Method findClosureOverload(Method method, Collection<Method> candidates) {
    Class<?>[] methodParameterTypes = method.getParameterTypes();
    return candidates.stream()
        .filter(candidate -> Arrays.equals(methodParameterTypes, candidate.getParameterTypes()))
        .findFirst()
        .orElse(null);
}
```

Reply:
Probably. As stated below, I don't want to change the code much.

Reply:
Done.